--- ./Utopia/source/trigger/trigger.c
+++ ./Utopia/source/trigger/trigger.c	2015-04-22 14:22:31.194837009 -0600
@@ -663,6 +663,8 @@
 
       long secs = get_next_timeout();
       if (-1 == secs) {
+	//zqiu: we still need to yeild to other process to avoid cpu occupation.
+	sleep(1);
          waitsecs = timeout.tv_sec = timeout.tv_usec = 0;
          rc = select(max_fd, &rd_set, NULL, NULL, NULL);
       } else {
