--- ./Utopia/source/scripts/init/c_registration/10_firewall.c
+++ ./Utopia/source/scripts/init/c_registration/10_firewall.c	2015-05-04 11:39:36.222783877 -0600
@@ -49,6 +49,8 @@
                                         "syslog_rotated|/etc/utopia/service.d/service_firewall/newhost_monitor.sh|NULL|"TUPLE_FLAG_EVENT,
                                         "ipv6_prefix|"SERVICE_DEFAULT_HANDLER,
                                         "current_wan_ipv6_interface|"SERVICE_DEFAULT_HANDLER,
+                                        "ipv6_wan0_dhcp_solicNodeAddr|"SERVICE_DEFAULT_HANDLER,
+                                        "ipv6_erouter0_dhcp_solicNodeAddr|"SERVICE_DEFAULT_HANDLER,
                                         NULL
                                       };
 
