--- ./Utopia/source/scripts/init/c_registration/02_lanHandler.c
+++ ./Utopia/source/scripts/init/c_registration/02_lanHandler.c	2015-06-15 07:21:49.136339225 -0600
@@ -33,7 +33,7 @@
 const char* SERVICE_DEFAULT_HANDLER = "/etc/utopia/service.d/lan_handler.sh";
 const char* SERVICE_CUSTOM_EVENTS[] = { 
     "snmp_subagent-status|/etc/utopia/service.d/lan_handler.sh",
-    "ipv4-resync|/etc/utopia/service.d/lan_handler.sh|"ACTION_FLAG_COLLAPSE_PENDING_QUEUE"|"TUPLE_FLAG_EVENT,
+    "ipv4-resync|/etc/utopia/service.d/lan_handler.sh|NULL|"TUPLE_FLAG_EVENT,
     "erouter_mode-updated|/etc/utopia/service.d/lan_handler.sh|NULL|"TUPLE_FLAG_EVENT,
     NULL };
 
